src/main/java/pw/itr0/kaba/encrypt/PBE.java

```diff
@@ -12,65 +14,207 @@
+     *
+     * @param salt           salt of PBE decryption
+     * @param iterationCount iteration count of PBE decryption
+     * @return PBE decrypter
+     */
+    public static Decrypter getDecrypter(byte[] salt, int iterationCount) {
+        byte[] copied = new byte[salt.length];
+        System.arraycopy(salt, 0, copied, 0, salt.length);
+        return new Decrypter(copied, iterationCount);
     }
 
-    private static Cipher getCipher(int mode, char[] password, byte[] salt, int iterationCount) throws NoSuchAlgorithmException,
-            InvalidKeySpecException, NoSuchPaddingException, InvalidAlgorithmParameterException, InvalidKeyException {
+    /**
+     * Returns {@link Cipher} for PBE encryption/decryption.
+     *
+     * @param mode           encryption({@link Cipher#ENCRYPT_MODE})/decryption({@link Cipher#DECRYPT_MODE})
+     * @param password       password of PBE
+     * @param salt           salt of PBE
+     * @param iterationCount iteration count of PBE
+     * @return {@link Cipher} for PBE encryption/decryption
+     * @throws NoSuchAlgorithmException if the running JRE does not support "{@value ENCRYPTION_ALGORITHM}"
+     * @throws InvalidKeyException      if the given key size exceeds the maximum allowable key size
+     */
+    private static Cipher getCipher(int mode, char[] password, byte[] salt, int iterationCount) throws NoSuchAlgorithmException, InvalidKeyException {
         PBEKeySpec keySpec = new PBEKeySpec(password);
         SecretKeyFactory keyFactory = SecretKeyFactory.getInstance(ENCRYPTION_ALGORITHM);
-        SecretKey key = keyFactory.generateSecret(keySpec);
+        SecretKey key;
+        try {
+            key = keyFactory.generateSecret(keySpec);
+        } catch (InvalidKeySpecException e) {
+            throw new MissImplementationException("Key specification must be type safely implemented. This exception must not occur.", e);
+        }
 
         PBEParameterSpec paramSpec = new PBEParameterSpec(salt, iterationCount);
 
-        Cipher cipher = Cipher.getInstance(ENCRYPTION_ALGORITHM);
-        cipher.init(mode, key, paramSpec);
+        Cipher cipher;
+        try {
+            cipher = Cipher.getInstance(ENCRYPTION_ALGORITHM);
+        } catch (NoSuchPaddingException e) {
+            throw new MissImplementationException("Padding algorithm must be statically implemented. This exception must not occur.", e);
+        }
+        try {
+            cipher.init(mode, key, paramSpec);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new MissImplementationException("Algorithm parameter must be type safely implemented. This exception must not occur.", e);
+        }
         return cipher;
     }
 
-    public static class Encrypter implements ByteEncrypter {
-
+    /**
+     * Encrypter for PBE.
+     *
+     * @author ryotan
+     * @since 1.0.0
+     */
+    public static class Encrypter implements PasswordBasedEncrypter {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class Encrypter should be declared as final.](https://www.codacy.com/app/ryotan/kaba-vault/file/2578430306/issues/source?bid=3298589&fileBranchId=3301306#l130)